src/Illuminate/Filesystem/FilesystemServiceProvider.php

```diff
@@ -60,6 +71,49 @@ protected function registerManager()
         });
     }
 
+    /**
+     * Register protected file serving.
+     *
+     * @return void
+     */
+    protected function serveFiles()
+    {
+        if ($this->app->routesAreCached()) {
+            return;
+        }
+
+        foreach ($this->app['config']['filesystems.disks'] ?? [] as $disk => $config) {
+            if (! $this->shouldServeFiles($config)) {
+                continue;
+            }
+
+            $this->app->booted(function () use ($disk, $config) {
+                $uri = isset($config['url'])
+                    ? rtrim(parse_url($config['url'])['path'], '/')
+                    : '/storage';
+
+                Route::get($uri.'/{path}', function (Request $request, string $path) use ($disk, $config) {
+                    return (new ServeFile(
+                        $disk,
+                        $config,
+                        $this->app->isProduction()
+                    ))($request, $path);
```

Comment:
Would it be possible to resolve this via the container? That would allow for easier modification.

Reply:
I believe you could use `Storage::serveUsing(fn)`